tests/unit/lib/utils/test_hash.py

```diff
@@ -122,3 +123,11 @@ def test_dir_cyclic_links(self):
     def test_str_checksum(self):
         checksum = str_checksum("Hello, World!")
         self.assertEqual(checksum, "65a8e27d8879283831b664bd8b7f0ad4")
+
+    @patch("samcli.lib.utils.hash.hashlib")
+    def test_md5_instantiation(self, patched_hashlib):
+        str_checksum("dummy-data")
+        if sys.version_info.major >= 3 & sys.version_info.minor >= 9:
+            patched_hashlib.md5.assert_called_with(usedforsecurity=False)
+        else:
+            patched_hashlib.md5.assert_called_with()
```

Comment:
Seems like python 3.9 is not compatible with this logic?

Reply:
Interesting, this branch is executed instead of above in Py3.9

Reply:
Looks like the condition is using the bitwise `&` instead of `and`

Reply:
Good catch, writing in different languages have messed my mind :)